### PR TITLE
add README & makefile lbraries

### PR DESCRIPTION
**Pull request by natebrennand**
This introduces the master copies of the makefile libraries that have been in review on appasaurus.

### CODE REVIEW

File: make/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+This takes advantage of several less commonly used make features
+
+- [the `include` directive](http://www.gnu.org/software/make/manual/make.html#Include)
+- [the `define` directive](http://www.gnu.org/software/make/manual/make.html#Multi_002dLine)
+- [the `eval` function](http://www.gnu.org/software/make/manual/make.html#Eval-Function)
+
+[This blog post](http://make.mad-scientist.net/the-eval-function/) provides a nice overview of how these can all be leveraged together.
+
+
+## Golang
```
Comment (mohit) [inline make/README.md]: Example Makefiles where these are used?

Comment (natebrennand) [inline make/README.md]: :+1: 
I'll make a PR on gearcmd so I can link a public repo

**Comment by natebrennand** [comment]
@mo: added a link to the gearcmd Makefile. Will work once https://github.com/Clever/gearcmd/pull/35 is merged

File: make/golang.mk [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+@echo "VETTING $(1)..."
+@go vet $(GOPATH)/src/$(1)/*.go
+endef
+
+# golang-test-all-deps installs all dependencies needed for different test cases.
+golang-test-all-deps: golang-fmt-deps golang-lint-deps golang-test-deps golang-vet-deps
+
+# golang-test-all calls fmt, lint, vet and test on the specified pkg.
+# arg1: pkg path
+define golang-test-all
```
Comment (jiancheung) [inline make/golang.mk]: I kind of wonder if only the `test-all`'s should be exported and all the individual stuff (such as `golang-lint`) should be kept "private". I haven't thought about it too much, just a passing comment.

Comment (natebrennand) [inline make/golang.mk]: I kept them separate with the use-case that an integration test endpoint might want to use fmt/lint/vet but not actually run the test outright.
Also there's no real notion of private AFAIK. Maybe we could precede it with an underscore?

Might be worth revisiting in 2-3 months and making them private if no one is using those targets

**Pull request merged**